homeassistant/components/tessie/__init__.py

```diff
@@ -0,0 +1,69 @@
+"""Tessie integration."""
+from datetime import timedelta
+from http import HTTPStatus
+import logging
+
+from aiohttp import ClientConnectionError, ClientResponseError
+from tessie_api import get_state_of_all_vehicles
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+TESSIE_SYNC_INTERVAL = 15
+PLATFORMS = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Tessie config."""
+    api_key = entry.data[CONF_ACCESS_TOKEN]
+    session = async_get_clientsession(hass)
+
+    async def async_get():
+        try:
+            vehicles = await get_state_of_all_vehicles(
+                session=session, api_key=api_key, only_active=False
+            )
+        except ClientResponseError as e:
+            if e.status == HTTPStatus.UNAUTHORIZED:
+                raise ConfigEntryAuthFailed() from e
+            raise UpdateFailed from e
+        except ClientConnectionError as e:
+            raise UpdateFailed from e
+        return {
+            vehicle["vin"]: vehicle["last_state"] for vehicle in vehicles["results"]
+        }
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="Tessie",
+        update_method=async_get,
+        update_interval=timedelta(seconds=TESSIE_SYNC_INTERVAL),
+    )
```

Comment:
Personally, I'd recommend splitting this out into it's own class. It will give you more flexibility in the future and helps with readability. But I don't think it is required.